post_office/tests/test_mail.py

```diff
@@ -395,4 +396,32 @@ def test_send_bulk_with_faulty_template(self):
                                      template=template, status=STATUS.queued)
         _send_bulk([email], uses_multiprocessing=False)
         email = Email.objects.get(id=email.id)
-        self.assertEqual(email.status, STATUS.failed)
\ No newline at end of file
+        self.assertEqual(email.status, STATUS.failed)
+
+    def test_retry_failed(self):
+        previous_settings = settings.POST_OFFICE
+        setattr(settings, 'POST_OFFICE', {'MAX_RETRIES': 2})
+
+        self.assertEqual(get_max_retries(), 2)
+        self.assertEqual(get_time_delta_to_retry(), timedelta(minutes=15))
+
+
+        email = Email.objects.create(to='to@example.com', from_email='from@example.com', status=STATUS.queued, backend_alias='error')
+        _send_bulk([email], uses_multiprocessing=False)
+        email = Email.objects.get(id=email.id)
+
+        self.assertEqual(int(email.scheduled_time.timestamp()), int(now().__add__(timedelta(minutes=15)).timestamp()))
+        self.assertEqual(email.status, STATUS.requeued)
+
+        email.scheduled_time = now()
+        email.save()
+        _send_bulk([email], uses_multiprocessing=False)
+        self.assertEqual(email.status, STATUS.requeued)
+
+        email.scheduled_time = now()
+        email.save()
+
+        _send_bulk([email], uses_multiprocessing=False)
+        self.assertEqual(email.status, STATUS.requeued)
+
```

Comment:
Can you add another call to `_send_bulk()`? After this call, `email.status` should be set to `failed`

Reply:
It's ok :D